dazel/lib/src/bazelify/build_command.dart

```diff
@@ -0,0 +1,142 @@
+import 'dart:async';
+import 'dart:io' hide exitCode;
+
+import 'package:archive/archive.dart';
+import 'package:args/command_runner.dart';
+import 'package:path/path.dart' as p;
+
+import 'arguments.dart';
+import 'build.dart';
+import 'exceptions.dart';
+
+class BuildCommand extends Command {
+  BuildCommand() {
+    argParser
+      ..addOption('app',
+          help: 'The name of the app target to build, this is the name of the '
+              'html file for the app. This argument may be provided as a '
+              'positional argument as well.')
+      ..addOption('output-dir',
+          abbr: 'o',
+          defaultsTo: 'deploy',
+          help: 'The directory to output build outputs to.');
+  }
+
+  @override
+  String get name => 'build';
+
+  @override
+  String get description => 'Builds a dart web app.';
+
+  final watch = new Stopwatch();
+
+  @override
+  Future<Null> run() async {
+    assert(!watch.isRunning);
+    watch.start();
+    var commonArgs = await sharedArguments(globalResults);
+    if (commonArgs == null) return;
+
+    var app = argResults['app'] as String;
+    // Support providing `app` as a positional argument.
+    if (app == null && argResults.rest.length == 1) {
+      app = argResults.rest.first;
+    } else {
+      throw new ApplicationFailedException(
+          'Missing required argument `app`:\n${argParser.usage}', 1);
+    }
+    // Target name doesn't have `.html`, but we want support for that for users.
+    if (app.endsWith('.html')) app = p.withoutExtension(app);
+
+    var buildArgs = new BazelifyBuildArguments._(
+        bazelExecutable: commonArgs.bazelExecutable,
+        pubPackageDir: commonArgs.pubPackageDir,
+        outputDir: argResults['output-dir'],
+        target: app);
+
+    await build(buildArgs);
+    watch
+      ..reset()
+      ..stop();
+  }
+
+  Future build(BazelifyBuildArguments args) async {
```

Comment:
None of the logic is duplicated anywhere else, and its not heavily nested, so I would prefer to leave it as is.